tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,575 @@
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png",
+    html_favicon_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png"
+)]
+
+use clap::{App, Arg};
+use std::collections::{HashMap, HashSet};
+use std::fs;
+use std::hash::Hash;
+use std::io::BufRead;
+use std::path::Path;
+use std::sync::Mutex;
+use syn::punctuated::Punctuated;
+use syn::{visit::Visit, ItemEnum};
+use syn::{Attribute, Field, ItemStruct, Lit, Meta, MetaNameValue, Type, UseTree, Variant};
+use walkdir::WalkDir;
+
+// unfortunately many global vars, since i cant add extra arguments to trait methods
+lazy_static::lazy_static! {
+    // When iterating recursively, this keeps track of the current path
+    static ref CURRENT_PATH: Mutex<Vec<TypeAndField>> = Mutex::new(vec![]);
+    // a set of CURRENT_PATH whenever it hits a pii=true field
+    static ref PII_TYPES: Mutex<HashSet<Vec<TypeAndField>>> = Mutex::new(HashSet::new());
+    // All the structs and enums, where the key is the full path. The string in the tuple represents
+    // the module path to the item. Now that I think of it, should be able to just get it from the key.
+    static ref ALL_TYPES: Mutex<HashMap<String, (EnumOrStruct, String)>> = Mutex::new(HashMap::new());
+
+    #[derive(Debug)]
+    static ref PII_VALUES: Mutex<Vec<String>> = Mutex::new(vec![]);
+}
+
+#[derive(PartialEq, Eq, Hash, Clone, Debug, Default)]
+struct TypeAndField {
+    type_name: String, //shoud be full path
+    field_name: String,
+}
+
+fn find_rs_files(dir: &str) -> Vec<std::path::PathBuf> {
+    let walker = WalkDir::new(dir).into_iter();
+    let mut rs_files = Vec::new();
+
+    for entry in walker.filter_map(walkdir::Result::ok) {
+        if entry.clone().path().to_string_lossy().contains("target/") {
+            continue;
+        }
+        if entry.file_type().is_file() && entry.path().extension().unwrap_or_default() == "rs" {
+            rs_files.push(entry.into_path());
+        }
+    }
+    rs_files
+}
+
+#[derive(Debug, Clone)]
+enum EnumOrStruct {
+    ItemStruct(ItemStruct),
+    ItemEnum(ItemEnum),
+}
```

Comment:
Related to my comment above, I'd change this to `TokenType`. We can also simplify the enum options by removing `Item`:
```suggestion
enum EnumOrStruct {
    Struct(ItemStruct),
    Enum(ItemEnum),
}
```

Reply:
good call, I was unnecessarily focused on namespaces here